H2TauTau/python/proto/analyzers/TreeVariables.py

```diff
@@ -150,15 +151,27 @@ def __init__(self, name, function=None, type=float):
     # JAN - only one PU mva working point, but we may want to add more
     # run in our skimming step
     # (for which Jet.py would have to be touched again)
-    Variable('mva_pu', lambda jet : jet.puMva('pileupJetId:fullDiscriminant')),
-    Variable('id_loose', lambda jet : jet.looseJetId()),
-    Variable('id_pu', lambda jet : jet.puJetId()),
-    Variable('mva_btag', lambda jet : jet.btagMVA),
-    Variable('area', lambda jet : jet.jetArea()),
-    Variable('flavour_parton', lambda jet : jet.partonFlavour()),
-    Variable('csv', lambda jet : jet.btagMVA),
-    Variable('rawfactor', lambda jet : jet.rawFactor()),
-    Variable('genjet_pt', lambda jet : jet.matchedGenJet.pt() if hasattr(jet, 'matchedGenJet') and jet.matchedGenJet else -999.),
+    Variable('mva_pu'                   , lambda jet : jet.puMva('pileupJetId:fullDiscriminant') if hasattr(jet, 'puMva'                    ) else -999.                      ),
+    Variable('id_loose'                 , lambda jet : jet.looseJetId()                          if hasattr(jet, 'looseJetId'               ) else -999.                      ),
+    Variable('id_pu'                    , lambda jet : jet.puJetId()                             if hasattr(jet, 'puJetId'                  ) else -999.                      ),
+    Variable('mva_btag'                 , lambda jet : jet.btagMVA                               if hasattr(jet, 'btagMVA'                  ) else -999.                      ),
+    Variable('area'                     , lambda jet : jet.jetArea()                             if hasattr(jet, 'jetArea'                  ) else -999.                      ),
+    Variable('flavour_parton'           , lambda jet : jet.partonFlavour()                       if hasattr(jet, 'partonFlavour'            ) else -999.                      ),
+    Variable('csv'                      , lambda jet : jet.btagMVA                               if hasattr(jet, 'btagMVA'                  ) else -999.                      ),
+    Variable('rawfactor'                , lambda jet : jet.rawFactor()                           if hasattr(jet, 'rawFactor'                ) else -999.                      ),
+    Variable('genjet_pt'                , lambda jet : jet.matchedGenJet.pt()                    if hasattr(jet, 'matchedGenJet'            ) and jet.matchedGenJet else -999.),
+    Variable('nConstituents'            , lambda jet : jet.nConstituents()                       if hasattr(jet, 'nConstituents'            ) else -999.                      ),
```

Comment:
these were useful for the tau ES BDT training, but in general they're just redundant.
Let me implement the switch you suggest